src/repl/repl_commands.cpp

```
@@ -1088,35 +1132,41 @@ bool dispatch_command(const std::string& command, const std::string& arg, ReplSt
-    // Compress history: keep system prompt + generate summary of conversation
+    // Compress history: keep system prompt + LLM-generated summary
     if (s.history.size() <= 2) {
       s.out << "[nothing to compress — conversation too short]\n";
     } else {
-      // Build summary from conversation
-      std::string summary = "Previous conversation summary: ";
-      int msg_count = 0;
+      s.out << "Compressing conversation...\n";
+      s.out.flush();
+      // Build the conversation text for summarization
+      std::string conversation;
```

Clang (14.0.6)

[warning] variable 'conversation' is not initialized

(cppcoreguidelines-init-variables)